File: backend-php/src/db/models/query.php
<?php

namespace Db\Models;

class Query {
    public $id;
    public $patientId;
    public $doctorId;
    public $question;
    public $response;
    public $status; // "pending", "answered", "closed"
    public $createdAt;
    public $updatedAt;
    public $answeredAt;
    public $urgency; // "low", "medium", "high"

    public function toArray() {
        return [
            'id' => $this->id,
            'patientId' => $this->patientId,
            'doctorId' => $this->doctorId,
            'question' => $this->question,
            'response' => $this->response,
            'status' => $this->status,
            'createdAt' => $this->createdAt,
            'updatedAt' => $this->updatedAt,
            'answeredAt' => $this->answeredAt,
            'urgency' => $this->urgency,
        ];
    }
}

class QueryWithUserInfo {
    public $id;
    public $patientId;
    public $patientName;
    public $doctorId;
    public $doctorName;
    public $question;
    public $response;
    public $status;
    public $urgency;
    public $createdAt;
    public $updatedAt;
    public $answeredAt;

    public function toArray() {
        return [
            'id' => $this->id,
            'patientId' => $this->patientId,
            'patientName' => $this->patientName,
            'doctorId' => $this->doctorId,
            'doctorName' => $this->doctorName,
            'question' => $this->question,
            'response' => $this->response,
            'status' => $this->status,
            'urgency' => $this->urgency,
            'createdAt' => $this->createdAt,
            'updatedAt' => $this->updatedAt,
            'answeredAt' => $this->answeredAt,
        ];
    }
}

File: backend/internal/db/models/query.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Query represents a question from a patient and a response from a doctor
type Query struct {
	ID         primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	PatientID  primitive.ObjectID `bson:"patientId" json:"patientId"`
	DoctorID   primitive.ObjectID `bson:"doctorId,omitempty" json:"doctorId,omitempty"`
	Question   string             `bson:"question" json:"question"`
	Response   string             `bson:"response,omitempty" json:"response,omitempty"`
	Status     string             `bson:"status" json:"status"` // "pending", "answered", "closed"
	CreatedAt  time.Time          `bson:"createdAt" json:"createdAt"`
	UpdatedAt  time.Time          `bson:"updatedAt" json:"updatedAt"`
	AnsweredAt *time.Time         `bson:"answeredAt,omitempty" json:"answeredAt,omitempty"`
	Urgency    string             `bson:"urgency,omitempty" json:"urgency,omitempty"` // "low", "medium", "high"
}

// QueryWithUserInfo represents a query with additional user information
type QueryWithUserInfo struct {
	ID          primitive.ObjectID `json:"id"`
	PatientID   primitive.ObjectID `json:"patientId"`
	PatientName string             `json:"patientName"`
	DoctorID    primitive.ObjectID `json:"doctorId,omitempty"`
	DoctorName  string             `json:"doctorName,omitempty"`
	Question    string             `json:"question"`
	Response    string             `json:"response,omitempty"`
	Status      string             `json:"status"`
	Urgency     string             `json:"urgency,omitempty"`
	CreatedAt   time.Time          `json:"createdAt"`
	UpdatedAt   time.Time          `json:"updatedAt"`
	AnsweredAt  *time.Time         `json:"answeredAt,omitempty"`
}